### PR TITLE
Lunged Event

### PR DESCRIPTION
**Pull request by VALERA771**
"Fix" for #1734. We've tried

### CODE REVIEW

**Review by louis1706 — APPROVED** [review]

**Comment by louis1706** [comment]
i miss something keep , IDeniableEvent and add TODO to remove it

**Comment by ITeMbI4** [comment]
If this event called after, why you can't name it `LungedEventArgs`, and can you please add `ev.Target` property to it?

**Comment by ITeMbI4** [comment]
@louis1706 why you approve this?
TriggerLunge called before, not after...

**Comment by ITeMbI4** [comment]
And what about https://github.com/Exiled-Team/EXILED/pull/1754 ?

**Comment by VALERA771** [comment]
IsAllowed there cause desync

**Comment by VALERA771** [comment]
> i miss something keep , IDeniableEvent and add TODO to remove it

Maybe just remove IDeniableEvent?

**Comment by Misaka-ZeroTwo** [comment]
Why are you changing the lunging event but not adding another lunged event?

**Comment by Misaka-ZeroTwo** [comment]
> Like make this ev args and event obsolete and make lunged?

Don't touch anything on lunging event, just do make another lunged event like you did before in #1709

**Comment by Misaka-ZeroTwo** [comment]
Dunno why you are putting things in obsolete, cuz #1754 will fix the lunging event.

**Comment by VALERA771** [comment]
> Dunno why you are putting things in obsolete, cuz #1754 will fix the lunging event.

Same thing I done in #1709

**Comment by Misaka-ZeroTwo** [comment]
> Same thing I done in #1709. And if it will fix I gonna just close this PR cuz they're gonna be same

No, the purpose of Lunged event is that it invokes after the event. Just like Dying and Died. And they can be used with different purposes.

**Comment by Misaka-ZeroTwo** [comment]
Also you could add targets in Lunged eventarg

**Comment by VALERA771** [comment]
> Also you could add targets in Lunged eventarg

Then it'll be just like hurting, isn't it?

**Comment by VALERA771** [comment]
Yes. If I add target it'll be just like hurting or dying ev

**Comment by Misaka-ZeroTwo** [comment]
> Yes. If I add target it'll be just like hurting or dying ev


It can just be get only.

**Comment by VALERA771** [comment]
Okay, let it be like this

**Review by louis1706 — CHANGES_REQUESTED** [review]
can you add a Void Prefix for Lunging (with Obsolete IsAllowed)

**Review by louis1706 — APPROVED** [review]

**Comment by joker-119** [comment]
Fix merge conflicts

File: Exiled.Events/EventArgs/Scp939/LungedEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LungedEventArgs"/> class.
+        /// </summary>
+        /// <param name="player"><inheritdoc cref="Player"/></param>
+        /// <param name="target"><inheritdoc cref="Target"/></param>
+        public LungedEventArgs(ReferenceHub player, ReferenceHub target)
+        {
+            Player = Player.Get(player);
+            Target = target == null ? null : Player.Get(target);
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Scp939/LungedEventArgs.cs:26]: Player.Get(null) should return null; so the check is useless

**Review by louis1706 — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Events/Scp939/Lunge.cs [hunk 2]
```diff
@@ -7,52 +7,23 @@
...
-                new(OpCodes.Call, Method(typeof(Scp939), nameof(Scp939.OnLunging))),
-            });
-
-            for (int z = 0; z < newInstructions.Count; z++)
-            {
-                yield return newInstructions[z];
-            }
-
-            ListPool<CodeInstruction>.Pool.Return(newInstructions);
+            LungingEventArgs ev = new(Player.Get(__instance.Owner));
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Events/Scp939/Lunge.cs:25]: Lunged?

File: Exiled.Events/Patches/Events/Scp939/Lunge.cs [hunk 3]
```diff
@@ -7,52 +7,23 @@
...
+            Handlers.Scp939.OnLunging(ev);
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Events/Scp939/Lunge.cs:26]: same as above

**Review by Misaka-ZeroTwo — CHANGES_REQUESTED** [review]

File: Exiled.Events/EventArgs/Scp939/LungingEventArgs.cs [hunk 4]
```diff
@@ -23,12 +23,9 @@ public class LungingEventArgs : IPlayerEvent, IDeniableEvent
         /// <param name="player">
         ///     <inheritdoc cref="Player" />
         /// </param>
-        /// <param name="isAllowed">
-        ///     <inheritdoc cref="IsAllowed" />
-        /// </param>
-        public LungingEventArgs(ReferenceHub player, bool isAllowed = true)
+        public LungingEventArgs(Player player)
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/EventArgs/Scp939/LungingEventArgs.cs:26]: breaking change. No need to change anything for LungingEventArgs,

**Pull request closed without merging**